components/Tags.tsx

```diff
@@ -1,53 +1,98 @@
-import { useEffect } from "react";
+import { isDefined } from "helpers/tinaHelpers";
+import { ReactElement, useEffect } from "react";
 import { resetPaginationPage } from "redux/slices/pagination";
 import { resetTags, setTags } from "redux/slices/selectedTags";
 import { useAppDispatch, useAppSelector } from "redux/typesHooks";
+import { useEditState } from "tinacms/dist/react";
+
+export interface TagData {
+  readonly tagName: string;
```

Comment:
Why read only?

Reply:
```suggestion
  readonly name: string;
```
`tag` is already in the object name

Reply:
> Why read only?

Because this value won't change, but if it does, there will be an error from typescript. `readonly` is a useful thing, I recommend you to use it as much as possible in your types, because it gains you confidence, that you don't mutate your code